File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/domain/WordDomainModel.java
package com.przemyslawlusnia.vocabularycreator.wordlist.domain;

import com.przemyslawlusnia.vocabularycreator.core.Constants;
import org.immutables.value.Value;

@Value.Immutable
public abstract class WordDomainModel {

  @Value.Default
  public String getWord() {
    return "";
  }

  @Value.Default
  public String getTranslation() {
    return "";
  }

  @Value.Default
  public int getType() {
    return Constants.TYPE_TRAINING;
  }

}

File: app/src/main/java/com/przemyslawlusnia/vocabularycreator/wordlist/domain/WordDomainModel.kt
package com.przemyslawlusnia.vocabularycreator.wordlist.domain

import com.przemyslawlusnia.vocabularycreator.core.Constants

data class WordDomainModel(val word: String = "",
                      val translation: String = "",
                      val type: Int = Constants.TYPE_TRAINING)